src/var/mutation_counting.jl

```diff
@@ -27,128 +60,168 @@ The strictest test of ambiguity, if either nucleotide is not
 A, T, G, or C, then this function returns true.
 """
 @inline function is_ambiguous_strict{T<:Nucleotide}(a::T, b::T)
-    return isambiguous(a) || isambiguous(b)
+    return isambiguous(a) | isambiguous(b)
```

Comment:
It is intentional. The reason is in julia the logical operators using double symbols e.g. "||" have short-circuit behaviour, which when used in loops can prevent the julia compiler from emitting simd instructions for that loop. To ensure straight-line code that can be vectorized using simd, the `|` version of the operator is used which ensures both sides of the argument are evaluated first.